packages/apputils-extension/src/settingconnector.ts

```diff
@@ -40,5 +40,9 @@ export class SettingConnector extends DataConnector<
     };
   }
 
+  async save(id: string, raw: string): Promise<void> {
+    await this._connector.save(id, raw);
```

Comment:
I have a slight preference toward `return this._connector.save(id, raw);` but not a strong one. What do you think?

Reply:
I was going to say "I'm happy to change it just because you asked, @afshin", but then I looked and saw that the function it's wrapping is also `async`:

https://github.com/jupyterlab/jupyterlab/blob/a1fc14a5971c780a385ed994c1fa99e9d3ed6730/packages/services/src/setting/index.ts#L99-L99

so for consistency's sake, I think `save` should stick with `async/await` here.

To be 100% sure, the `async` and `return Promise` versions are semantically equivalent: both will immediately return a `Promise` that resolves upon completion of the wrapped function, right?

Reply:
Is there an extra promise getting wrapped up in here? Basically I'm wondering if these three are equivalent or if they return a different level of nesting.
```typescript
save(id: string, raw: string): Promise<void> {
  return this._connector.save(id, raw);
}
```
```typescript
async save(id: string, raw: string): Promise<void> {
  return this._connector.save(id, raw);
}
```
```typescript
async save(id: string, raw: string): Promise<void> {
  await this._connector.save(id, raw);
}
```

Reply:
I've read over the [`async/await` docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function) and I've live inspected all three versions in DevTools. Here's my conclusions:

- From the perspective of a caller, 1, 2, and 3 are identical. All return a singly nested promise that resolves to an `undefined`.

- 1 and 2 are actually identical. If a function is already returning a `Promise`, all adding the `async` keyword does is allow for the use of `await` within the function body.

- 3 *might* involve one extra step in it's evaluation. Basically, 1/2 just return the inner promise, whereas 3 returns a promise that resolves when the *wrapper* function finishes. The extra step would then be the implicit `return void 0` that the wrapper function calls after it awaits the inner promise.

So at least I can confidently say that no, no there's no extra promise getting wrapped up here. I can also say that for our purposes there are no important distinctions between 1/2/3. @afshin Do you still think I should change it?

Reply:
Okay, so I think if we are exporting ES2015, **1** is the best answer, based on this experiment. Does that look right to you? You can play with the playground link to confirm.

<img width="1433" alt="Screen Shot 2019-12-10 at 21 37 27" src="https://user-images.githubusercontent.com/159529/70571217-49c5d380-1b95-11ea-87f4-dbd5f1dc481b.png">


https://www.typescriptlang.org/play/?target=2#code/MYGwhgzhAEBCkFNoG8BQ0PUgTwHbGgDMB7YgCgEoAuaABQCdiBbASwgQB4A3YlgEwB8KdJlFgA7mBYAXOo1bsAdPQQRiILgjIBycaWnaKAbhEYAvqgupQkGAGFmAB2Ls+w0cGK4I0+gFdgaWJ6MgAjRBp4dgp3UUxpAAs2RQB9cPZoAF5odIQTUSsxCDwCEnJqOWY2Th5+ITQ4zBVpP3pcaETktMRFMsp8zELMcJCKhir2bl5BWMbm1vbOiFTc3tJ+02ghjBx8HLAAL0oacYUa6frNsUkZDqTl7qU+403t6Ed6Fi4waSRHhEiiBMZiAA

Reply:
(This is all very pedantic, but it's good to know what JS we are actually generating.)